azurerm/internal/services/containers/container_registry_resource.go

```diff
@@ -309,19 +377,32 @@ func resourceArmContainerRegistryUpdate(d *schema.ResourceData, meta interface{}
 	oldGeoReplicationLocations := old.(*schema.Set)
 	newGeoReplicationLocations := new.(*schema.Set)
 
+	// handle upgrade to Premium SKU first
+	if skuChange && isPremiumSku {
+		if err := applyContainerRegistrySku(d, meta, sku, resourceGroup, name); err != nil {
+			return fmt.Errorf("Error applying sku %q for Container Registry %q (Resource Group %q): %+v", sku, name, resourceGroup, err)
+		}
+	}
```

Comment:
instead of dealing with the SKU spereately like this, could we maybe just do the policies afterwards ensuring any sku changes are done first?

Reply:
re read PR description, please disregard as this makes sense now 🙂